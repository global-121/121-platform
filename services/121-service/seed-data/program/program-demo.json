{
  "published": false,
  "validation": true,
  "phase": "design",
  "location": "Demoland",
  "ngo": "Demo NGO",
  "titlePortal": {
    "en": "Demo Cash Aid Project"
  },
  "titlePaApp": {
    "en": "Demo Cash Aid Project"
  },
  "description": {
    "en": ""
  },
  "startDate": "2022-09-29T12:00:00Z",
  "endDate": "2030-11-30T12:00:00Z",
  "currency": "EUR",
  "distributionFrequency": "week",
  "distributionDuration": 10,
  "fixedTransferValue": 35,
  "financialServiceProviders": [
    {
      "fsp": "Intersolve-voucher-whatsapp",
      "configuration": [
        {
          "name": "username",
          "value": "INTERSOLVE_USERNAME"
        },
        {
          "name": "password",
          "value": "INTERSOLVE_PASSWORD"
        }
      ]
    },
    {
      "fsp": "Africas-talking"
    }
  ],
  "targetNrRegistrations": 870,
  "tryWhatsAppFirst": true,
  "meetingDocuments": {
    "en": ""
  },
  "phoneNumberPlaceholder": "+31 600 00 00 00",
  "programCustomAttributes": [],
  "programQuestions": [
    {
      "name": "name",
      "label": {
        "en": "1. What is your name?"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Name"
      }
    },
    {
      "name": "middlename",
      "label": {
        "en": "2. What is your middle name?"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Middle name"
      }
    },
    {
      "name": "firstname",
      "label": {
        "en": "3. What is your first name?"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "First name"
      }
    },
    {
      "name": "gender",
      "label": {
        "en": "4. What is your gender?"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "Masculin ",
          "label": {
            "en": "Male",
            "fr": "Masculin"
          }
        },
        {
          "option": "Féminin",
          "label": {
            "en": "Female",
            "fr": "Féminin"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Gender"
      }
    },
    {
      "name": "birthdate",
      "label": {
        "en": "5. What is your birth date?"
      },
      "answerType": "date",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Birth date"
      }
    },
    {
      "name": "documentid",
      "label": {
        "en": "6. What is your document ID number? (- of none)"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "duplicateCheck": true,
      "phases": ["registrationValidation", "inclusion", "payment"],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Document ID"
      }
    },
    {
      "name": "headhousehold",
      "label": {
        "en": "7. Are you the head of household?"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "Oui",
          "label": {
            "en": "Yes",
            "fr": "Oui"
          }
        },
        {
          "option": "Non",
          "label": {
            "en": "No",
            "fr": "Non"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Household head"
      }
    },
    {
      "name": "males",
      "label": {
        "en": "8. How many male(s) of 18 or older live in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Men"
      }
    },
    {
      "name": "boys",
      "label": {
        "en": "9. How many boy(s) of 17 or younger live in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 1
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Boys"
      }
    },
    {
      "name": "women",
      "label": {
        "en": "10. How many women of 18 or older live in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Women"
      }
    },
    {
      "name": "girls",
      "label": {
        "en": "11. How many girl(s) of 17 or younger live in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 1
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Girls"
      }
    },
    {
      "name": "householdsituation",
      "label": {
        "en": "12. What is your household's situation?"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "returnee",
          "label": {
            "en": "Retournee",
            "fr": "Retourné"
          }
        },
        {
          "option": "displaced",
          "label": {
            "en": "Displaced",
            "fr": "Déplacé"
          }
        },
        {
          "option": "autochtoneprivate",
          "label": {
            "en": "Autochtone (private residency)",
            "fr": "Autochtone (simple résident)"
          }
        },
        {
          "option": "autochtonehost",
          "label": {
            "en": "Autochtone (host family)",
            "fr": "Autochtone (famille d’accueil)"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "returnee": 2,
        "displaced": 2,
        "autochtoneprivate": 0,
        "autochtonehost": 1
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Household situation"
      }
    },
    {
      "name": "maritalsituation",
      "label": {
        "en": "13. What is the marital situation of the head of your household?"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "widowed",
          "label": {
            "en": "Widowed",
            "fr": "Veuf/Veuve"
          }
        },
        {
          "option": "married",
          "label": {
            "en": "Married",
            "fr": "Marié(e)"
          }
        },
        {
          "option": "abandoned",
          "label": {
            "en": "Abandoned",
            "fr": "Abandonné(e)"
          }
        },
        {
          "option": "seperated",
          "label": {
            "en": "Seperated",
            "fr": "Séparé(e)"
          }
        },
        {
          "option": "divorced",
          "label": {
            "en": "Divorced",
            "fr": "Divorcé (e)"
          }
        },
        {
          "option": "girl",
          "label": {
            "en": "Girl/adolescent < 18 pregnant or mother",
            "fr": "Enfant/adolescent ménage"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "widowed": 1,
        "married": 0,
        "abandoned": 1,
        "seperated": 1,
        "divorced": 1,
        "girl": 3
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Marital situation"
      }
    },
    {
      "name": "specialcharacteristics",
      "label": {
        "en": "14. What are special characteristics of the head of your household (if any)?"
      },
      "answerType": "multi-select",
      "questionType": "standard",
      "options": [
        {
          "option": "olderalone",
          "label": {
            "en": "Older person living alone"
          }
        },
        {
          "option": "elderlycharge",
          "label": {
            "en": "Elderly person in charge of a household"
          }
        },
        {
          "option": "pregnantbrestfeeding",
          "label": {
            "en": "Pregnant/breastfeeding woman"
          }
        },
        {
          "option": "womancharge",
          "label": {
            "en": "Woman in charge of a household"
          }
        },
        {
          "option": "childcharge",
          "label": {
            "en": "Child/adolescent in charge of a household"
          }
        },
        {
          "option": "girlmother",
          "label": {
            "en": "Girl/adolescent < 18 pregnant or mother"
          }
        },
        {
          "option": "none",
          "label": {
            "en": "None"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "olderalone": 0,
        "elderlycharge": 1,
        "pregnantbrestfeeding": 2,
        "womancharge": 1,
        "childcharge": 3,
        "girlmother": 4,
        "none": 0
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Special characteristics"
      }
    },
    {
      "name": "pregnant",
      "label": {
        "en": "15. How many pregnant women are in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 1
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Pregnant women"
      }
    },
    {
      "name": "breastfeeding",
      "label": {
        "en": "16. How many breastfeeding women are in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 1
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Breastfeeding women"
      }
    },
    {
      "name": "womendisabled",
      "label": {
        "en": "17. How many women in your household aged 18 and over have a disability? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 2
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Women disability"
      }
    },
    {
      "name": "mendisabled",
      "label": {
        "en": "18. How many men in your household aged 18 and over have a disability? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 2
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Men disability"
      }
    },
    {
      "name": "girlsdisabled",
      "label": {
        "en": "19. How many girls in your household aged under 18 have a disability? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 2
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Girls disability"
      }
    },
    {
      "name": "boysdisabled",
      "label": {
        "en": "20. How many boys in your household aged under 18 have a disability? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 2
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Boys disability"
      }
    },
    {
      "name": "childrenmalnourished",
      "label": {
        "en": "21. How many malnourished children are in your household? (0 if none)"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {
        "multiplier": 2
      },
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Malnourished children"
      }
    },
    {
      "name": "malnourishedtreatment",
      "label": {
        "en": "22. If there are malnourished children in your household are you pursuing treatment at the health center?"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "Oui",
          "label": {
            "en": "Yes",
            "fr": "Oui"
          }
        },
        {
          "option": "Non",
          "label": {
            "en": "No",
            "fr": "Non"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Malnourished treatment"
      }
    }
  ],
  "aboutProgram": {
    "en": "[about aid program]"
  },
  "fullnameNamingConvention": ["name", "middlename", "firstname"],
  "languages": ["en"]
}
