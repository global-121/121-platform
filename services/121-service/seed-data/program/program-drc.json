{
  "published": true,
  "validation": true,
  "phase": "design",
  "location": "Democratic Republic of the Congo",
  "ngo": "Help a Child",
  "titlePortal": {
    "en": "Help a Child Digital Aid DRC"
  },
  "titlePaApp": {
    "en": "Help a Child DRC Cash Support"
  },
  "description": {
    "en": ""
  },
  "startDate": "2020-09-01T12:00:00Z",
  "endDate": "2022-11-30T12:00:00Z",
  "currency": "USD",
  "distributionFrequency": "once",
  "distributionDuration": 1,
  "fixedTransferValue": 53,
  "financialServiceProviders": [
    {
      "fsp": "VodaCash"
    }
  ],
  "targetNrRegistrations": 870,
  "tryWhatsAppFirst": false,
  "phoneNumberPlaceholder": "+24 300 00 00 00",
  "programCustomAttributes": [],
  "programQuestions": [
    {
      "name": "lastName",
      "label": {
        "en": "B.2. Name",
        "fr": "B.2. Nom de famille"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true
    },
    {
      "name": "middleName ",
      "label": {
        "en": "B.3. Middle name",
        "fr": "B.3. Deuxième prénom"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true
    },
    {
      "name": "firstName",
      "label": {
        "en": "B.4. First name",
        "fr": "B.4. Prénom"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true
    }
  ],
  "aboutProgram": {
    "en": "This <strong>Help a Child DRC</strong> aid program was created to support people who are in need. Help a Child DRC will distribute money via VodaCom to included people.",
    "fr": "Le programme d'aide Help a Child DRC a été créé pour soutenir les personnes dans le besoin. Help a Child DRC distribuera de l'argent via VodaCom aux personnes concernées."
  },
  "fullnameNamingConvention": ["lastName", "middleName", "firstName"],
  "languages": ["en", "fr"],
  "enableMaxPayments": false,
  "enableScope": false,
  "allowEmptyPhoneNumber": false
}
