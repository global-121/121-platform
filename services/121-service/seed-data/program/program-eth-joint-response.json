{
  "published": false,
  "validation": false,
  "phase": "registrationValidation",
  "location": "Demoland",
  "ngo": "Demo NGO",
  "titlePortal": {
    "en": "Demo Cash Aid Project"
  },
  "titlePaApp": {
    "en": "Demo Cash Aid Project"
  },
  "description": {
    "en": ""
  },
  "startDate": "2022-09-29T12:00:00Z",
  "endDate": "2030-11-30T12:00:00Z",
  "currency": "ETB",
  "distributionFrequency": "month",
  "distributionDuration": 6,
  "fixedTransferValue": 1.0,
  "financialServiceProviders": [
    {
      "fsp": "Commercial-bank-ethiopia"
    }
  ],
  "targetNrRegistrations": 870,
  "tryWhatsAppFirst": false,
  "meetingDocuments": {
    "en": ""
  },
  "bankAccountNumberPlaceholder": "1000263499216",
  "programCustomAttributes": [],
  "programQuestions": [
    {
      "name": "fullname",
      "label": {
        "en": "What is your first name?"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "First name"
      }
    },
    {
      "name": "gender",
      "label": {
        "en": "4. What is your gender?"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "Masculin ",
          "label": {
            "en": "Male",
            "fr": "Masculin"
          }
        },
        {
          "option": "Féminin",
          "label": {
            "en": "Female",
            "fr": "Féminin"
          }
        }
      ],
      "persistence": true,
      "export": ["all-people-affected", "included", "selected-for-validation"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "shortLabel": {
        "en": "Gender"
      }
    }
  ],
  "aboutProgram": {
    "en": "[about aid program]"
  },
  "fullnameNamingConvention": ["fullname"],
  "languages": ["en"],
  "enableMaxPayments": true
}
