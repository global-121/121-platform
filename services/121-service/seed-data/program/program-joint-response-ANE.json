{
  "published": false,
  "validation": false,
  "phase": "design",
  "location": "Ethiopia",
  "ngo": "ANE",
  "titlePortal": {
    "en": "DRA Joint Response 2023 Ethiopia - ANE",
    "et_AM": "የዲአርኤ የጋራ ምላሽ 2023 ኢትዮጵያ - አኔ"
  },
  "titlePaApp": {
    "en": "DRA Joint Response 2023 Ethiopia - ANE",
    "et_AM": "የዲአርኤ የጋራ ምላሽ 2023 ኢትዮጵያ - አኔ"
  },
  "description": {
    "en": ""
  },
  "startDate": "2023-08-01T12:00:00Z",
  "endDate": "2023-12-31T12:00:00Z",
  "currency": "ETB",
  "distributionFrequency": "month",
  "distributionDuration": 3,
  "fixedTransferValue": 7000,
  "tryWhatsAppFirst": false,
  "paymentAmountMultiplierFormula": null,
  "financialServiceProviders": [
    {
      "fsp": "Commercial-bank-ethiopia"
    }
  ],
  "targetNrRegistrations": 250,
  "meetingDocuments": {
    "en": ""
  },
  "phoneNumberPlaceholder": "+251 000 000 000",
  "programCustomAttributes": [],
  "programQuestions": [
    {
      "name": "fullName",
      "label": {
        "en": "Name",
        "et_AM": "ስም"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "persistence": true,
      "export": ["all-people-affected", "included"],
      "scoring": {},
      "phases": [],
      "editableInPortal": true,
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "phoneNumber",
      "label": {
        "en": "Phone number",
        "et_AM": "ስልክ ቁጥር"
      },
      "answerType": "tel",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "idNumber",
      "label": {
        "en": "ID Card number",
        "et_AM": "የመታውቂያ ቁጥር"
      },
      "answerType": "text",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": true,
      "placeholder": null
    },
    {
      "name": "age",
      "label": {
        "en": "Age",
        "et_AM": "እድሜ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "gender",
      "label": {
        "en": "Gender",
        "et_AM": "ጾታ"
      },
      "answerType": "dropdown",
      "questionType": "standard",
      "options": [
        {
          "option": "male",
          "label": {
            "en": "Male",
            "et_AM": "ወንድ"
          }
        },
        {
          "option": "female",
          "label": {
            "en": "Female",
            "et_AM": "ሴት"
          }
        }
      ],
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyFemale",
      "label": {
        "en": "How many female?",
        "et_AM": "ሴት አባላት"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyMale",
      "label": {
        "en": "How many male?",
        "et_AM": "ወንድ አባላት"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "totalFamilyMembers",
      "label": {
        "en": "Total family members",
        "et_AM": "በቤተሰብዎ ውስጥ ያሉት አባላት"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyFemaleUnder18",
      "label": {
        "en": "Female under 18",
        "et_AM": "ሴት ከ18 ዓመት በታች"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyMaleUnder18",
      "label": {
        "en": "Male under 18",
        "et_AM": "ወንድ ከ18 ዓመት በታች"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyFemaleOver18",
      "label": {
        "en": "Female over 18",
        "et_AM": "ሴት ከ18 ዓመት በላይ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyMaleOver18",
      "label": {
        "en": "Male over 18",
        "et_AM": "ወንድ ከ18 ዓመት በላይ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyFemaleDisabilityUnder18",
      "label": {
        "en": "Female with disability under 18",
        "et_AM": "ሴት ከ18 ዓመት በታች አካል ጉዳተኛ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyMaleDisabilityUnder18",
      "label": {
        "en": "Male with disability under 18",
        "et_AM": "ወንድ ከ18 ዓመት በታች አካል ጉዳተኛ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyFemaleDisabilityOver18",
      "label": {
        "en": "Female with disability over 18",
        "et_AM": "ሴት ከ18 ዓመት በላይ አካል ጉዳተኛ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    },
    {
      "name": "howManyMaleDisabilityOver18",
      "label": {
        "en": "Male with disability over 18",
        "et_AM": "ወንድ ከ18 ዓመት በላይ አካል ጉዳተኛ"
      },
      "answerType": "numeric",
      "questionType": "standard",
      "options": null,
      "scoring": {},
      "persistence": true,
      "pattern": null,
      "phases": [],
      "editableInPortal": true,
      "export": ["all-people-affected", "included"],
      "duplicateCheck": false,
      "placeholder": null
    }
  ],
  "aboutProgram": {
    "en": "DRA Joint Response Ethiopia 2023 - ANE"
  },
  "fullnameNamingConvention": ["fullName"],
  "languages": ["en", "et_AM"],
  "enableMaxPayments": true,
  "enableScope": false,
  "allowEmptyPhoneNumber": false
}
