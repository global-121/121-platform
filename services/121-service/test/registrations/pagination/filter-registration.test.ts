import { RegistrationStatusEnum } from '../../../src/registration/enum/registration-status.enum';
import { SeedScript } from '../../../src/scripts/seed-script.enum';
import { ProgramPhase } from '../../../src/shared/enum/program-phase.enum';
import { changePhase } from '../../helpers/program.helper';
import {
  awaitChangePaStatus,
  getRegistrations,
  importRegistrations,
} from '../../helpers/registration.helper';
import { getAccessToken, resetDB } from '../../helpers/utility.helper';
import {
  createExpectedValueObject,
  expectedAttributes,
  programIdOCW,
  registrationOCW1,
  registrationOCW2,
  registrationOCW3,
  registrationOCW4,
} from './pagination-data';

describe('Load PA table', () => {
  describe('getting registration using paginate', () => {
    let accessToken: string;

    beforeAll(async () => {
      await resetDB(SeedScript.nlrcMultiple);
      accessToken = await getAccessToken();

      await changePhase(
        programIdOCW,
        ProgramPhase.registrationValidation,
        accessToken,
      );

      await importRegistrations(
        programIdOCW,
        [
          registrationOCW1,
          registrationOCW2,
          registrationOCW3,
          registrationOCW4,
        ],
        accessToken,
      );

      await awaitChangePaStatus(
        programIdOCW,
        [registrationOCW1.referenceId],
        RegistrationStatusEnum.included,
        accessToken,
      );
    });

    it('should filter based on status', async () => {
      // Act
      const getRegistrationsResponse = await getRegistrations(
        programIdOCW,
        null,
        accessToken,
        null,
        null,
        { 'filter.status': RegistrationStatusEnum.included },
      );
      const data = getRegistrationsResponse.body.data;
      const meta = getRegistrationsResponse.body.meta;

      // Assert
      expect(data[0]).toMatchObject(
        createExpectedValueObject(registrationOCW1, 1),
      );
      for (const attribute of expectedAttributes) {
        expect(data[0]).toHaveProperty(attribute);
      }
      expect(meta.totalItems).toBe(1);
    });

    it('should filter based on registration data', async () => {
      // Act
      const getRegistrationsResponse = await getRegistrations(
        programIdOCW,
        null,
        accessToken,
        null,
        null,
        { 'filter.whatsappPhoneNumber': registrationOCW4.whatsappPhoneNumber },
      );
      const data = getRegistrationsResponse.body.data;
      const meta = getRegistrationsResponse.body.meta;

      // Assert
      expect(data[0]).toMatchObject(
        createExpectedValueObject(registrationOCW4, 4),
      );
      for (const attribute of expectedAttributes) {
        expect(data[0]).toHaveProperty(attribute);
      }
      expect(meta.totalItems).toBe(1);
    });

    it('should filter based on root attributes & registration data', async () => {
      // Act
      const getRegistrationsResponse = await getRegistrations(
        programIdOCW,
        null,
        accessToken,
        null,
        null,
        {
          'filter.whatsappPhoneNumber': registrationOCW3.whatsappPhoneNumber,
          'filter.preferredLanguage': registrationOCW3.preferredLanguage,
        },
      );
      const data = getRegistrationsResponse.body.data;
      const meta = getRegistrationsResponse.body.meta;

      // Assert
      expect(data[0]).toMatchObject(
        createExpectedValueObject(registrationOCW3, 3),
      );
      for (const attribute of expectedAttributes) {
        expect(data[0]).toHaveProperty(attribute);
      }
      expect(meta.totalItems).toBe(1);
    });

    it('should filter using in, eq, ilike and null', async () => {
      // Act
      // Each of the filters would seperately return
      const getRegistrationsResponse = await getRegistrations(
        programIdOCW,
        null,
        accessToken,
        null,
        null,
        {
          'filter.whatsappPhoneNumber': `$ilike:${registrationOCW3.whatsappPhoneNumber.substring(
            0,
            1,
          )}`,
          'filter.preferredLanguage': `$in:nonExisting,${registrationOCW3.preferredLanguage}`,
          'filter.addressCity': `$eq:${registrationOCW3.addressCity}`,
        },
      );
      const data = getRegistrationsResponse.body.data;
      const meta = getRegistrationsResponse.body.meta;

      // Assert
      expect(data[0]).toMatchObject(
        createExpectedValueObject(registrationOCW3, 3),
      );
      for (const attribute of expectedAttributes) {
        expect(data[0]).toHaveProperty(attribute);
      }
      expect(meta.totalItems).toBe(1);
    });

    it('should filter using search in combination with filter', async () => {
      // Act
      // The postal code shoud filter 1 and 2 and the search should filter 2 and 4, so only 2 should be returned
      const getRegistrationsResponse = await getRegistrations(
        programIdOCW,
        null,
        accessToken,
        null,
        null,
        {
          'filter.addressPostalCode': `$ilike:${registrationOCW2.addressPostalCode.substring(
            0,
            1,
          )}`,
          search: `${registrationOCW2.addressCity}`,
        },
      );
      const data = getRegistrationsResponse.body.data;
      const meta = getRegistrationsResponse.body.meta;

      // Assert
      expect(data[0]).toMatchObject(
        createExpectedValueObject(registrationOCW2, 2),
      );
      for (const attribute of expectedAttributes) {
        expect(data[0]).toHaveProperty(attribute);
      }
      expect(meta.totalItems).toBe(1);
    });
  });
});
