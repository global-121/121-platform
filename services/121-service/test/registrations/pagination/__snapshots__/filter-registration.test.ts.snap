// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Filter registrations not operator filters should throw bad request for $not:$null filter on registration attribute data 1`] = `"Using $not:$null is not supported for registration attribute data filters."`;
